src/main/java/seedu/address/model/person/Company.java

```diff
@@ -0,0 +1,56 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's company in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidCompany(String)}
+ */
+public class Company {
+
+    public static final String MESSAGE_CONSTRAINTS = "Companies can take any values, and it should not be blank";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[^\\s].*";
```

Comment:
Is there a reason we are using this regex for all the new fields? Or do we need to update it later?

Reply:
I've used the most general regex that just makes sure its not a blank string for the time being. I was thinking when we finalise our command syntax we can fine tune the validation patterns. If you think there's a better pattern for now I'll update.

Reply:
AFAIK, this regex disallows the single-line string to start with '\' or 's'? Correct me if I'm wrong

Reply:
Not super familiar with patterns but this is what I gather from looking around:
\s represents a whitespace character, ^ means not, so the first character has to be not whitespace.
. means any character and * means repeating 0 or more times so it can be any string of any length.

Reply:
Okay so it basically works similar to the `trim()` function, but only removes trailing spaces from the front and not the back of the string. We can keep it for now then, thanks!